mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -35,21 +47,34 @@ def _hit_to_mlflow_experiment(self, hit: Any) -> Experiment:
                           artifact_location=hit.artifact_location,
                           lifecycle_stage=hit.lifecycle_stage)
 
-    def _hit_to_mlflow_run(self, hit: Any) -> Run:
+    def _hit_to_mlflow_run(self, hit: Any, use_latest_metrics: bool = False) -> Run:
         return Run(run_info=self._hit_to_mlflow_run_info(hit),
-                   run_data=self._hit_to_mlflow_run_data(hit))
+                   run_data=self._hit_to_mlflow_run_data(hit, use_latest_metrics))
 
     def _hit_to_mlflow_run_info(self, hit: Any) -> RunInfo:
         return RunInfo(run_uuid=hit.meta.id, run_id=hit.meta.id,
-                       experiment_id=str(hit.experiment_id), user_id=hit.user_id,
-                       status=hit.status, start_time=hit.start_time,
+                       experiment_id=str(hit.experiment_id),
+                       user_id=hit.user_id,
+                       status=hit.status,
+                       start_time=hit.start_time,
                        end_time=hit.end_time if hasattr(hit, 'end_time') else None,
                        lifecycle_stage=hit.lifecycle_stage, artifact_uri=hit.artifact_uri)
 
-    def _hit_to_mlflow_run_data(self, hit: Any) -> RunData:
-        return RunData(metrics=[self._hit_to_mlflow_metric(m) for m in hit.metrics],
-                       params=[self._hit_to_mlflow_param(p) for p in hit.params],
-                       tags=[self._hit_to_mlflow_tag(t) for t in hit.tags])
+    def _hit_to_mlflow_run_data(self, hit: Any, use_latest_metrics: bool) -> RunData:
+        if use_latest_metrics:
+            return RunData(metrics=[self._hit_to_mlflow_metric(m) for m in
+                                    (hit.latest_metrics if hasattr(hit, 'latest_metrics') else [])],
+                           params=[self._hit_to_mlflow_param(p) for p in
+                                   (hit.params if hasattr(hit, 'params') else [])],
+                           tags=[self._hit_to_mlflow_tag(t) for t in
+                                 (hit.tags if hasattr(hit, 'tags') else[])])
+        else:
+            return RunData(metrics=[self._hit_to_mlflow_metric(m) for m in
+                                    (hit.metrics if hasattr(hit, 'metrics') else [])],
+                           params=[self._hit_to_mlflow_param(p) for p in
+                                   (hit.params if hasattr(hit, 'params') else [])],
+                           tags=[self._hit_to_mlflow_tag(t) for t in
+                                 (hit.tags if hasattr(hit, 'tags') else[])])
```

Comment:
You don't need to duplicate the code for params and tags, just handle the metrics before instantiating the RunData